phpspy.c

```diff
@@ -796,6 +801,55 @@ void log_error(const char *fmt, ...) {
     }
 }
 
+static int get_comm_pid(pid_t pid, char *buf, size_t buf_size) {
+    char proc_fs_path[PATH_MAX];
+    char *line;
+    ssize_t fread_res;
+    size_t comm_len;
+    size_t line_len = 0;
+
+    FILE *f = NULL;
+    snprintf(proc_fs_path, PATH_MAX, "/proc/%d/comm", pid);
+
+    f = fopen(proc_fs_path, "rb");
+    if (f == NULL) {
+        log_error("Couldn't read comm of pid %d - errno %d\n", pid, errno);
+        goto error;
+    }
+
+    fread_res = getline(&line, &line_len, f);
```

Comment:
You don't initialize `line` before reading?

And why use a complicated API like `getline` when you just gotta read 16 bytes?